examples/structured_2d_dgsem/elixir_advection_basic_coupled.jl

```diff
@@ -76,30 +80,39 @@ semi = SemidiscretizationCoupled((semi1, semi2))
 # ODE solvers, callbacks etc.
 
 # Create ODE problem with time span from 0.0 to 1.0
-ode = semidiscretize(semi, (0.0, 1.0));
+ode = semidiscretize(semi, (0.0, 2.0));
 
 # At the beginning of the main loop, the SummaryCallback prints a summary of the simulation setup
 # and resets the timers
 summary_callback = SummaryCallback()
 
+# The AnalysisCallback allows to analyse the solution in regular intervals and prints the results
+analysis_callback1 = AnalysisCallback(semi1, interval=2)
+analysis_callback2 = AnalysisCallback(semi2, interval=2)
+analysis_callback = AnalysisCallbackCoupled(semi, analysis_callback1, analysis_callback2)
+
 # The SaveSolutionCallback allows to save the solution to a file in regular intervals
-save_solution = SaveSolutionCallback(interval=100,
+save_solution = SaveSolutionCallback(interval=2,
                                      solution_variables=cons2prim)
 
 # The StepsizeCallback handles the re-calculation of the maximum Δt after each time step
 stepsize_callback = StepsizeCallback(cfl=1.6)
 
 # Create a CallbackSet to collect all callbacks such that they can be passed to the ODE solver
-callbacks = CallbackSet(summary_callback, stepsize_callback, save_solution)
+callbacks = CallbackSet(summary_callback, analysis_callback, save_solution, stepsize_callback)
 
 
 ###############################################################################
 # run the simulation
 
 # OrdinaryDiffEq's `solve` method evolves the solution in time and executes the passed callbacks
-sol = solve(ode, CarpenterKennedy2N54(williamson_condition=false),
-            dt=1.0, # solve needs some value here but it will be overwritten by the stepsize_callback
-            save_everystep=false, callback=callbacks);
+# sol = solve(ode, CarpenterKennedy2N54(williamson_condition=false),
+#             dt=1.0, # solve needs some value here but it will be overwritten by the stepsize_callback
+#             save_everystep=false, callback=callbacks);
```

Comment:
@ranocha If I try to use the OrdinaryDiffEq version, I get the following error:
```
ERROR: LoadError: No matching function wrapper was found!
Stacktrace:
   [1-7] ⋮ internal
       @ FunctionWrappersWrappers, SciMLBase
     [8] macro expansion
       @ ~/.julia/packages/TimerOutputs/RsWnF/src/TimerOutput.jl:391 [inlined]
     [9] macro expansion
       @ /mnt/ssd/home/mschlott/hackathon/coupling-Trixi.jl/src/callbacks_step/analysis.jl:314 [inlined]
    [10] macro expansion
       @ /mnt/ssd/home/mschlott/hackathon/coupling-Trixi.jl/src/auxiliary/auxiliary.jl:274 [inlined]
    [11] (::AnalysisCallback{Trixi.LobattoLegendreAnalyzer{Float64, 7, …}, Tuple{typeof(Trixi.entropy_timederivative)}, …})(u_ode::SubArray{Float64, 1, …}, du_ode::SubArray{Float64, 1, …}, integrator::OrdinaryDiffEq.ODEIntegrator{CarpenterKennedy2N54{typeof(OrdinaryDiffEq.trivial_limiter!), typeof(OrdinaryDiffEq.trivial_limiter!), …}, true, …}, semi::SemidiscretizationHyperbolic{StructuredMesh{2, Float64}, LinearScalarAdvectionEquation2D{Float64}, …})
       @ Trixi /mnt/ssd/home/mschlott/hackathon/coupling-Trixi.jl/src/callbacks_step/analysis.jl:271
    [12] initialize!(cb::DiscreteCallback{Trixi.var"#1202#1210"{Int64}, AnalysisCallback{Trixi.LobattoLegendreAnalyzer{Float64, 7, …}, Tuple{typeof(Trixi.entropy_timederivative)}, …}, …}, u_ode::SubArray{Float64, 1, …}, du_ode::SubArray{Float64, 1, …}, t::Float64, integrator::OrdinaryDiffEq.ODEIntegrator{CarpenterKennedy2N54{typeof(OrdinaryDiffEq.trivial_limiter!), typeof(OrdinaryDiffEq.trivial_limiter!), …}, true, …}, semi::SemidiscretizationHyperbolic{StructuredMesh{2, Float64}, LinearScalarAdvectionEquation2D{Float64}, …})
       @ Trixi /mnt/ssd/home/mschlott/hackathon/coupling-Trixi.jl/src/callbacks_step/analysis.jl:211
    [13] initialize!(cb_coupled::DiscreteCallback{Trixi.var"#1208#1216"{Tuple{DiscreteCallback{Trixi.var"#1202#1210"{Int64}, AnalysisCallback{Trixi.LobattoLegendreAnalyzer{Float64, 7, …}, Tuple{typeof(Trixi.entropy_timederivative)}, …}, …}, DiscreteCallback{Trixi.var"#1202#1210"{Int64}, AnalysisCallback{Trixi.LobattoLegendreAnalyzer{Float64, 7, …}, Tuple{typeof(Trixi.entropy_timederivative)}, …}, …}}}, AnalysisCallbackCoupled{Tuple{DiscreteCallback{Trixi.var"#1202#1210"{Int64}, AnalysisCallback{Trixi.LobattoLegendreAnalyzer{Float64, 7, …}, Tuple{typeof(Trixi.entropy_timederivative)}, …}, …}, DiscreteCallback{Trixi.var"#1202#1210"{Int64}, AnalysisCallback{Trixi.LobattoLegendreAnalyzer{Float64, 7, …}, Tuple{typeof(Trixi.entropy_timederivative)}, …}, …}}}, …}, u_ode_coupled::Vector{Float64}, t::Float64, integrator::OrdinaryDiffEq.ODEIntegrator{CarpenterKennedy2N54{typeof(OrdinaryDiffEq.trivial_limiter!), typeof(OrdinaryDiffEq.trivial_limiter!), …}, true, …})
       @ Trixi /mnt/ssd/home/mschlott/hackathon/coupling-Trixi.jl/src/callbacks_step/analysis.jl:691
 [14-24] ⋮ internal
       @ DiffEqBase, OrdinaryDiffEq, Unknown
    [25] solve(prob::ODEProblem{Vector{Float64}, Tuple{Float64, Float64}, …}, args::CarpenterKennedy2N54{typeof(OrdinaryDiffEq.trivial_limiter!), typeof(OrdinaryDiffEq.trivial_limiter!), …}; sensealg::Nothing, u0::Nothing, p::Nothing, wrap::Val{true}, kwargs::Base.Pairs{Symbol, Any, …})
       @ DiffEqBase ~/.julia/packages/DiffEqBase/HoOGI/src/solve.jl:882
    [26] top-level scope
       @ /mnt/ssd/home/mschlott/hackathon/coupling-Trixi.jl/examples/structured_2d_dgsem/elixir_advection_basic_coupled.jl:109
    [27] include
       @ ./Base.jl:458 [inlined]
    [28] trixi_include(mod::Module, elixir::String; kwargs::Base.Pairs{Symbol, Union{}, …})
       @ Trixi /mnt/ssd/home/mschlott/hackathon/coupling-Trixi.jl/src/auxiliary/special_elixirs.jl:43
    [29] trixi_include
       @ /mnt/ssd/home/mschlott/hackathon/coupling-Trixi.jl/src/auxiliary/special_elixirs.jl:38 [inlined]
    [30] #trixi_include#1830
       @ /mnt/ssd/home/mschlott/hackathon/coupling-Trixi.jl/src/auxiliary/special_elixirs.jl:46 [inlined]
    [31] trixi_include(elixir::String)
       @ Trixi /mnt/ssd/home/mschlott/hackathon/coupling-Trixi.jl/src/auxiliary/special_elixirs.jl:46
    [32] top-level scope
       @ REPL[9]:1
Use `err` to retrieve the full stack trace.
in expression starting at /mnt/ssd/home/mschlott/hackathon/coupling-Trixi.jl/examples/structured_2d_dgsem/elixir_advection_basic_coupled.jl:109
```
Do you know where this is coming from?

Reply:
Thanks a lot, this works perfectly!